auth-table-based-service/src/main/java/io/stargate/auth/table/AuthTableBasedServiceActivator.java

```diff
@@ -0,0 +1,99 @@
+        this.context = context;
+        log.info("Starting authTableBasedService....");
+
+        synchronized (authTableBasedService) {
+            try {
+                context.addServiceListener(this, String.format("(Identifier=%s)", PERSISTENCE_IDENTIFIER));
+            } catch (InvalidSyntaxException ise) {
+                throw new RuntimeException(ise);
+            }
+
+            persistenceReference = context.getServiceReference(Persistence.class.getName());
+            if (persistenceReference != null && persistenceReference.getProperty("Identifier").equals(PERSISTENCE_IDENTIFIER)) {
+                log.info("Setting persistence in AuthTableBasedServiceActivator");
+                this.authTableBasedService.setPersistence((Persistence)context.getService(persistenceReference));
+            }
+
+            if (persistenceReference != null) {
+                log.info("Registering authTableBasedService in AuthTableBasedServiceActivator");
+                registration = context.registerService(AuthenticationService.class.getName(), authTableBasedService, props);
+            }
+        }
+    }
+
+    @Override
+    public void stop(BundleContext context) {
+        if (persistenceReference != null) {
+            context.ungetService(persistenceReference);
+        }
+        // Do not need to unregister the service, because the OSGi framework will automatically do so
+    }
+
+    @Override
+    public void serviceChanged(ServiceEvent serviceEvent) {
+        int type = serviceEvent.getType();
+        String[] objectClass = (String[]) serviceEvent.getServiceReference().getProperty("objectClass");
+        synchronized (authTableBasedService) {
+            switch (type) {
+                case (ServiceEvent.REGISTERED):
+                    log.info("Service of type " + objectClass[0] + " registered.");
+                    Object service = context.getService(serviceEvent.getServiceReference());
+
+                    if (service instanceof Persistence) {
+                        log.info("Setting persistence in RestApiActivator");
+                        this.authTableBasedService.setPersistence((Persistence) service);
+                    }
+
+                    if (this.authTableBasedService.getPersistence() != null && registration == null) {
+                        log.info("Registering authTableBasedService in AuthTableBasedServiceActivator");
+                        registration = context.registerService(AuthenticationService.class.getName(), authTableBasedService, props);
+                    }
+                    break;
+                case (ServiceEvent.UNREGISTERING):
+                    log.info("Service of type " + objectClass[0] + " unregistered.");
+                    context.ungetService(serviceEvent.getServiceReference());
+                    break;
+                case (ServiceEvent.MODIFIED):
+                    // TODO: [doug] 2020-06-15, Mon, 12:58 do something here...
```

Comment:
I saw this comment before in the C2 repo, out of curiosity what _should_ we be doing here? re-getting the service?